src/webgpu/web_platform/canvas/getCurrentTexture.spec.ts

```diff
@@ -260,3 +267,86 @@ g.test('resize')
     currentTexture = ctx.getCurrentTexture();
     t.expect(prevTexture === currentTexture);
   });
+
+g.test('expiry')
+  .desc(
+    `
+Test automatic WebGPU canvas texture expiry on all canvas types with the following requirements:
+- getCurrentTexture returns the same texture object within the same frame, throughout:
+  - after previous frame update the rendering
+  - before current frame update the rendering
+  - in a microtask off the current frame task
+- getCurrentTexture returns a new texture object and the old texture object becomes invalid
+  as soon as possible after HTML update the rendering.
+  `
+  )
+  .params(u =>
+    u //
+      .combine('canvasType', kAllCanvasTypes)
```

Comment:
We should also add cases for transferControlToOffscreen. There's a number of things to test for that case, so probably should just write them down and do another PR. Technically there is already a TODO to improve kAllCanvasTypes but I don't think it will cover all of these.

Cases:

- if on the same thread, expiry happens when the document updates its rendering (window "rPAF") OR transferToImageBitmap is called
- if on a different thread, expiry happens when the worker updates its rendering (worker "rPAF") OR transferToImageBitmap is called
- [draw, transferControlToOffscreen, then canvas is displayed] on either {main thread, or transferred to worker}
- [draw, canvas is displayed, then transferControlToOffscreen] on either {main thread, or transferred to worker}
- reftests for the above 2 (what gets displayed when the canvas is displayed?)

Also possible things to test for canvases generally (canvas element or transferControlToOffscreen):
- canvas is added to DOM after being rendered
- canvas is already in DOM but becomes visible after being rendered

Unfortunately we can't technically test anything (except transferToImageBitmap) when the canvas is not in the DOM because expiry is never guaranteed to happen.

Reply:
> when the canvas is not in the DOM
Is it that in cts we can only have these test in reftests?

Reply:
reftests compare screenshots so they're only useful for things that make it to the page.

Reply:
There are already tests for at least one of thseo

Also possible things to test for canvases generally (canvas element or transferControlToOffscreen):

> * canvas is added to DOM after being rendered
> * canvas is already in DOM but becomes visible after being rendered

It would be easy for me to add another case or 2.